src/control-plane/backend/lambda/listen-state-status/index.ts

```
@@ -0,0 +1,300 @@
+      TableName: clickStreamTableName,
+      Key: {
+        id: projectId,
+        type: `PIPELINE#${pipelineId}#latest`,
+      },
+      UpdateExpression: 'SET #executionDetail = :executionDetail',
+      ExpressionAttributeNames: {
+        '#executionDetail': 'executionDetail',
+      },
+      ExpressionAttributeValues: {
+        ':executionDetail': executionDetail,
+      },
+    };
+    await docClient.send(new UpdateCommand(input));
+  } catch (err) {
+    logger.error('Failed to update pipeline state status: ', { projectId, pipelineId, err });
+  }
+};
+
+const deleteProject = async (projectId: string) => {
+  try {
+    // Scan all project versions
+    const input: ScanCommandInput = {
+      TableName: clickStreamTableName,
+      FilterExpression: 'id = :p AND deleted = :d',
+      ExpressionAttributeValues: {
+        ':p': projectId,
+        ':d': false,
+      },
+    };
+    const records: Record<string, NativeAttributeValue>[] = [];
+    for await (const page of paginateScan({ client: docClient }, input)) {
+      records.push(...page.Items as Record<string, NativeAttributeValue>[]);
+    }
+    if (records.length === 0) {
+      return;
+    }
+    const transactInput: TransactWriteItemsCommandInput = {
+      TransactItems: [],
+    };
+    for (let rec of records) {
+      transactInput.TransactItems!.push({
+        Update: {
+          TableName: clickStreamTableName,
+          Key: {
+            id: { S: projectId },
+            type: { S: rec.type },
+          },
+          UpdateExpression: 'SET deleted= :d, statusType= :s ',
+          ExpressionAttributeValues: {
+            ':d': { BOOL: true },
+            ':s': { S: PipelineStatusType.DELETED },
+          },
+        },
+      });
+    }
+    const params: TransactWriteItemsCommand = new TransactWriteItemsCommand(transactInput);
+    await docClient.send(params);
+  } catch (err) {
+    logger.error('Failed to delete project: ', { projectId, err });
```

This swallows this event. I prefer to throw the exception to retry it later.